DOCSP-30511: Find usage example (#64)
mongodb/docs-rust: source/usage-examples/find.txt

filter matches documents whose value of the ``cuisine`` field is ``"French"``.

.. tabs::

   .. tab:: Asynchronous
      :tabid: find-async

      .. io-code-block::
         :copyable: true

        .. input:: /includes/usage-examples/code-snippets/find-async.rs
            :language: rust
            :dedent:

        .. output::
            :language: console
            :visible: false

            // Results truncated
            ...
            Restaurant { name: "Cafe Un Deux Trois", cuisine: "French" }
            Restaurant { name: "Calliope", cuisine: "French" }
            ...

   .. tab:: Synchronous
      :tabid: find-sync

      .. io-code-block::
         :copyable: true

        .. input:: /includes/usage-examples/code-snippets/find-sync.rs
            :language: rust
            :dedent:

        .. output::
            :language: console
            :visible: false

            // Results truncated
            ...
            Restaurant { name: "Cafe Un Deux Trois", cuisine: "French" }
            Restaurant { name: "Calliope", cuisine: "French" }
            ...

Additional Information
----------------------

> To learn more about retrieving documents, see the :ref:`rust-retrieve-guide` guide.
> 
> For more information about using cursors, see the :ref:`rust-cursor-guide` guide.
> 
> To learn more about using query filters, see the :ref:`rust-query-guide` guide.

Reviewer: S: for consistency, change the second item to use the same introductory clause
ie
```suggestion
To learn more about retrieving documents, see the :ref:`rust-retrieve-guide` guide.

To learn how to use cursors, see the :ref:`rust-cursor-guide` guide.

To learn more about using query filters, see the :ref:`rust-query-guide` guide.
```

.. _rust-usage-find:

=======================
Find Multiple Documents
=======================

You can query for multiple documents in a collection by calling the ``find()``
method on a ``Collection`` instance. 

> Pass a query filter to the ``find()`` method to query for and return documents
> that match the filter in the collection. If you do not include a filter, MongoDB

Reviewer: ```suggestion
Pass a query filter to the ``find()`` method to return documents in the collection
that match the filter. If you do not include a filter, MongoDB
```

returns all the documents in the collection.

> The ``find()`` method returns a ``Cursor`` type, which you can iterate through
> to retrieve the documents that match the filter criteria. For more information
> on iterating through documents returned by a ``Cursor``, see :ref:`rust-cursor-guide`.

Reviewer: ```suggestion
The ``find()`` method returns a ``Cursor`` type, which you can iterate through
to retrieve individual documents. For more information about using cursors, see :ref:`rust-cursor-guide`.
```

> on iterating through documents returned by a ``Cursor``, see :ref:`rust-cursor-guide`.

Reviewer: S: move the more information to the addtl info section

Example
-------

> This example finds documents from the ``restaurants`` collection that match a
> query filter. The example uses a ``Restaurants`` struct containing ``name`` and
> ``cuisine`` fields to model the documents in the collection. 

Reviewer: ```suggestion
This example finds documents from the ``sample_restaurants.restaurants`` collection that match a query
filter. The example uses a ``Restaurant`` struct that has ``name`` and
``cuisine`` fields to model the documents in the collection.
```

The following code passes the following parameters to the ``find()`` method:

> - A query filter that matches documents containing a ``cuisine`` value of ``"French"``

Reviewer: ```suggestion
- A query filter that matches documents where the value of the ``cuisine`` field is ``"French"``
```

- An options parameter that sorts the matching documents by their ``"name"`` values in

>   alphabetical order

Reviewer: S: If you remove the sort, remove this point and make this part a sentence instead of a list

.. tabs::

   .. tab:: Asynchronous
      :tabid: find-async

      .. io-code-block::
         :copyable: true

        .. input:: /includes/usage-examples/code-snippets/find-async.rs
            :language: rust
            :dedent:

        .. output::
            :language: console
            :visible: false

            // Results truncated
            ...
            Restaurant { name: "Cafe Un Deux Trois", cuisine: "French" }
            Restaurant { name: "Calliope", cuisine: "French" }
            ...

   .. tab:: Synchronous
      :tabid: find-sync

      .. io-code-block::
         :copyable: true

        .. input:: /includes/usage-examples/code-snippets/find-sync.rs
            :language: rust
            :dedent:

        .. output::
            :language: console
            :visible: false

            // Results truncated
            ...
            Restaurant { name: "Cafe Un Deux Trois", cuisine: "French" }
            Restaurant { name: "Calliope", cuisine: "French" }
            ...

Additional Information
----------------------

To learn more about retrieving documents, see the :ref:`rust-retrieve-guide` guide.

> To learn more about using query filters, see :ref:`rust-query-guide`.

Reviewer: ```suggestion
To learn more about using query filters, see the :ref:`rust-query-guide` guide.
```

API Documentation

> -----------------

Reviewer: S: make this a subheading
